android/src/main/java/com/infitio/adharasocketio/AdharaSocketIoPlugin.java

```diff
@@ -19,12 +20,12 @@
 /**
  * AdharaSocketIoPlugin
  */
-public class AdharaSocketIoPlugin implements MethodCallHandler {
+public class AdharaSocketIoPlugin implements FlutterPlugin,MethodCallHandler {
 
   SparseArray<AdharaSocket> instances;
   private int currentIndex;
-  private final Registrar registrar;
-  private final StreamsChannel streamsChannel;
+  private FlutterPluginBinding flutterPluginBinding;
+  private StreamsChannel streamsChannel;
```

Comment:
reason for `streamsChannel` not being `final`?

Reply:
The new Android embedding 2.0 calls the default constructor initially (very annoying) so we can not initializer the stream channel. The onAttachedToEngine calls our private constructor which initializes StreamChannel. If the default constructor did not get called or we set a dummy value for it ion this constructor then that would work but is not very elegant.